File: manga-translator-backend/services/common/schemas/project.py
from __future__ import annotations
"""项目相关 Pydantic Schema"""
from typing import Optional, List
from pydantic import BaseModel, Field


class ProjectCreate(BaseModel):
    """创建项目请求"""
    name: str = Field(..., min_length=1, max_length=200, description="作品名称")
    source_lang: str = Field(..., min_length=1, max_length=10, description="源语言")
    cover_url: Optional[str] = Field(None, max_length=500, description="封面URL")


class ProjectUpdate(BaseModel):
    """更新项目请求"""
    name: Optional[str] = Field(None, min_length=1, max_length=200)
    source_lang: Optional[str] = Field(None, min_length=1, max_length=10)
    cover_url: Optional[str] = Field(None, max_length=500)
    is_favorite: Optional[bool] = None
    status: Optional[str] = Field(None, max_length=20)


class ProjectResponse(BaseModel):
    """项目响应"""
    project_id: str
    user_id: str
    name: str
    source_lang: str
    cover_url: Optional[str] = None
    is_favorite: bool = False
    status: str = "active"
    trashed_at: Optional[str] = None
    created_at: Optional[str] = None
    updated_at: Optional[str] = None
    chapter_count: Optional[int] = Field(default=0, description="章节数量")

    class Config:
        from_attributes = True


class ProjectListQuery(BaseModel):
    """项目列表查询参数"""
    page: int = Field(default=1, ge=1, description="页码")
    page_size: int = Field(default=20, ge=1, le=100, description="每页条数")
    status: Optional[str] = Field(None, description="状态筛选: active/trashed")
    search: Optional[str] = Field(None, max_length=200, description="名称搜索")
    sort_by: Optional[str] = Field(default="created_at", description="排序字段")
    sort_order: Optional[str] = Field(default="desc", description="排序方向: asc/desc")


class ProjectBatchRequest(BaseModel):
    """批量操作请求"""
    project_ids: List[str] = Field(..., min_length=1, description="项目ID列表")

File: manga-translator-web/src/types/project.ts
// 项目相关类型定义

export type SourceLang = 'ja' | 'zh' | 'en' | 'ko';
export type ProjectStatus = 'active' | 'trashed';

export interface ProjectData {
  project_id: string;
  user_id: string;
  name: string;
  source_lang: SourceLang;
  default_target_lang?: string;
  cover_url?: string;
  is_favorite: boolean;
  status: ProjectStatus;
  trashed_at?: string;
  created_at: string;
  updated_at: string;
  // 聚合字段
  chapter_count?: number;
  page_count?: number;
  completed_count?: number;
}

export interface CreateProjectParams {
  name: string;
  source_lang: SourceLang;
  default_target_lang?: string;
}

export interface UpdateProjectParams {
  name?: string;
  cover_url?: string;
  is_favorite?: boolean;
}

export interface ChapterData {
  chapter_id: string;
  project_id: string;
  name: string;
  sort_order: number;
  created_at: string;
  updated_at: string;
  page_count?: number;
}

export interface CreateChapterParams {
  name: string;
}

export interface UpdateChapterParams {
  name?: string;
  sort_order?: number;
}
